airflow/www/static/js/openDatasetModal.js

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* global $, isoDateToTimeEl, document */
+
+import { getMetaValue } from './utils';
+
+function openDatasetModal(dagId, summary) {
+  let nextRunUrl = getMetaValue('next_run_datasets_url');
+  $('#datasets_tbody').empty();
+  $('#datasets_error').hide();
+  $('#datasetNextRunModal').modal({});
+  $('#dag_id').text(dagId);
+  $('#next_run_summary').text(summary);
+  $('#datasets-loading-dots').css('display', 'inline-block');
+  if (dagId) {
+    if (nextRunUrl.includes('__DAG_ID__')) {
+      nextRunUrl = nextRunUrl.replace('__DAG_ID__', dagId);
+    }
+    $.get(nextRunUrl)
+      .done(
+        (datasets) => {
+          datasets.forEach((d) => {
+            const row = document.createElement('tr');
+
+            const uriCell = document.createElement('td');
+            const datasetLink = document.createElement('a');
+            datasetLink.href = `/datasets?dataset_id=${d.id}`;
```

Comment:
This won't be the right URL if the Airflow isn't "rooted" at `/` (and this is why we use the uri-in-metavalue trick elsewhere)